### PR TITLE
Support links as indicated in JSON:API specification

### PR DESCRIPTION
**Pull request by mqchau**
Solves #16 

Click [here](https://jsonapi.org/format/#document-resource-object-links) for the specifications related to links in JSON:API

This is my first time coding in Typescript so I may not use all the features it offers.

In order to use my new decorator, one would follow this:

```ts
@Model()
export class PersonWithLinks extends SpraypaintBase {
  static endpoint = "/v1/people_with_links"
  static jsonapiType = "people_with_links"
 
  @Link() self!: string
  @Link() webView!: string
}

const person = await PersonWithLinks.find(123)
// Assuming that the pay load is
// {
//   data: {
//     id: "123",
//     type: "people_with_links",
//     links: {
//       self: "/v1/people_with_links/123",
//       web_view: "/people_with_links/123"
//     }
//   }
// }

person.links.self // value is /v1/people_with_links/123
person.links.webView // value is /people_with_links/123
```

### CODE REVIEW

**Comment by richmolj** [comment]
This PR is looking great, thanks @mqchau! ❤️ 

This would actually serve as the foundation for something higher-level. This does a great job of giving *access* to the links but doesn't *follow* them. I'm thinking of a python client I wrote (not open source) that has an interface like

```ts
let { data } = post.links('comments') // or comments.related

// same as something like

let { data } = await Comment.where({ post_id: 123 }).all()
```

This way spraypaint can lazy-load associations - otherwise we'd have a link but not a way to follow that link and return spraypaint model instances.

It would also be helpful to re-use scopes and merge in criteria:

```ts
// merge in additional scoping params
let { data } = post.links('comments', Comment.where({ likes: { gt: 5 } }))
```

All that said, my thought is we can merge this PR as-is, and build on top of it. I can see the lower-level use as valid, particularly for accessing metadata. @wadetandy just want to get your thoughts before moving this direction.

@mqchau FWIW, the `links` I personally generate all point to the API. Was there a reason for making something like `webView` a link instead of an attribute, considering it points to a non-API url?

**Comment by mqchau** [comment]
I follow Github API model where they puts bunch of web view links of the Pull Request into the API response of Pull Request model

https://developer.github.com/v3/pulls/#get-a-single-pull-request

Look at the part of 
```
 "_links": {
    "self": {
      "href": "https://api.github.com/repos/octocat/Hello-World/pulls/1347"
    },
    "html": {
      "href": "https://github.com/octocat/Hello-World/pull/1347"
    },
  }
```

**Comment by mqchau** [comment]
@wadetandy can you pleas take a look at this soon? I'm waiting on this to proceed with my project at work.

File: test/unit/decorators.test.ts [hunk 1]
```diff
@@ -157,6 +158,39 @@ describe("Decorators", () => {
...
+      class TestClass extends BaseModel {
+        @Link() link1!: string
+      }
+
+      class ChildTestClass extends TestClass {
+        @Link() link2!: string
+      }
+
+      expect(ChildTestClass.linkList).to.include("link1")
+      expect(ChildTestClass.linkList).to.include("link2")
```
Comment (wadetandy) [inline test/unit/decorators.test.ts]: I'm almost positive it doesn't, but would add an assertion that `link2` is not in the parent list.

Comment (mqchau) [reply test/unit/decorators.test.ts]: Sure, agree with that.

**Comment by mqchau** [comment]
So you want me to add my decorators to the test of es6 compatibility test?

**Comment by wadetandy** [comment]
Yeah, you can run them with [this package script](https://github.com/graphiti-api/spraypaint.js/blob/master/package.json#L25)

**Comment by mqchau** [comment]
I got to admit that was a tricky incompatibility. I think I got it fixed though. Thanks for watching out for that.

**Comment by wadetandy** [comment]
Thanks for the update. This looks great!

**Comment by richmolj** [comment]
Thanks again @mqchau, great work ❤️ !

**Comment by mqchau** [comment]
Thanks for reviewing guys. Can you guys publish it on npmjs.com please?

**Comment by wadetandy** [comment]
0.7.0 should be published in the next few minutes once CI completes.

**Pull request merged**